### ISSUE TITLE
Build with parallel feature disabled is broken in 0.9.3

### ISSUE THREAD

**Issue by Ralith** [amethyst/shred#169]
```
error[E0433]: failed to resolve: could not find `rayon` in `{{root}}`
 --> /home/ralith/.cargo/registry/src/github.com-1ecc6299db9ec823/shred-0.9.3/src/dispatch/dispatcher.rs:7:56
  |
7 | pub type ThreadPoolWrapper = Option<::std::sync::Arc<::rayon::ThreadPool>>;
  |                                                        ^^^^^ could not find `rayon` in `{{root}}`

error[E0609]: no field `thread_pool` on type `dispatch::builder::DispatcherBuilder<'a, 'b>`
   --> /home/ralith/.cargo/registry/src/github.com-1ecc6299db9ec823/shred-0.9.3/src/dispatch/builder.rs:238:28
    |
238 |         dispatcher_builder.thread_pool = self.thread_pool.clone();
    |                            ^^^^^^^^^^^ help: a field with a similar name exists: `thread_local`

error[E0609]: no field `thread_pool` on type `&mut dispatch::builder::DispatcherBuilder<'a, 'b>`
   --> /home/ralith/.cargo/registry/src/github.com-1ecc6299db9ec823/shred-0.9.3/src/dispatch/builder.rs:238:47
    |
238 |         dispatcher_builder.thread_pool = self.thread_pool.clone();
    |                                               ^^^^^^^^^^^
```
Building without default features should be tested in CI.

**Comment by azriel91** [comment]
Hiya, this is fixed by #163, released in 0.9.4.

**Comment by azriel91** [comment]
https://github.com/amethyst/shred/pull/170 should catch building without the `"parallel"` feature, as it includes:

```bash
cargo test --verbose --no-default-features --features "shred-derive"
```
